## Indian Premier League (IPL) Regression Analysis

In [1]:
# Connect to Google Drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# As usual, we begin by loading the packages we will need

import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf

In [3]:
# Now we load the data

IPL=pd.read_excel("/content/drive/MyDrive/Sports Performance Analytics Michigan/Course 1 - Foundations of Sports Analytics: Data, Representation, and Models in Sports/Week 5 Content/Week 5 Assignment/IPL (assignment) data.xlsx")

In [4]:
IPL

,year,team,played,won,lost,noresult,points,netrunrate,champions,runnersup,third,fourth,salaries
0,2008,Chennai Super Kings,14,8,6,0,16,-0.192,0,1,0,0,5825000
1,2009,Chennai Super Kings,14,8,5,1,17,0.951,0,0,0,0,6765000
2,2010,Chennai Super Kings,14,7,7,0,14,0.274,1,0,0,0,4890000
3,2011,Chennai Super Kings,14,9,5,0,18,0.443,1,0,0,0,6330000
4,2012,Chennai Super Kings,16,8,7,1,17,0.100,0,1,0,0,7900000
...,...,...,...,...,...,...,...,...,...,...,...,...,...
87,2014,Sunrisers Hyderabad,14,6,8,0,12,-0.399,0,0,0,0,9803333
88,2015,Sunrisers Hyderabad,14,7,7,0,14,-0.239,0,0,0,0,8104667
89,2016,Sunrisers Hyderabad,14,8,6,0,16,0.245,1,0,0,0,8747666
90,2017,Sunrisers Hyderabad,14,8,5,1,17,0.599,0,0,0,1,8200500


In [5]:
# Create the sum of salaries in each season
Sumsal = IPL.groupby(['year'])['salaries'].sum().reset_index().rename(columns={'salaries':'allsal'})
Sumsal

,year,allsal
0,2008,34105000
1,2009,33445000
2,2010,33075000
3,2011,62210000
4,2012,59706250
5,2013,81535000
6,2014,73973332
7,2015,65242665
8,2016,63483666
9,2017,63014833


In [6]:
# Merge the two dataframes on year
IPL = pd.merge(IPL, Sumsal, on=['year'], how='left')
display(IPL)

,year,team,played,won,lost,noresult,points,netrunrate,champions,runnersup,third,fourth,salaries,allsal
0,2008,Chennai Super Kings,14,8,6,0,16,-0.192,0,1,0,0,5825000,34105000
1,2009,Chennai Super Kings,14,8,5,1,17,0.951,0,0,0,0,6765000,33445000
2,2010,Chennai Super Kings,14,7,7,0,14,0.274,1,0,0,0,4890000,33075000
3,2011,Chennai Super Kings,14,9,5,0,18,0.443,1,0,0,0,6330000,62210000
4,2012,Chennai Super Kings,16,8,7,1,17,0.100,0,1,0,0,7900000,59706250
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
87,2014,Sunrisers Hyderabad,14,6,8,0,12,-0.399,0,0,0,0,9803333,73973332
88,2015,Sunrisers Hyderabad,14,7,7,0,14,-0.239,0,0,0,0,8104667,65242665
89,2016,Sunrisers Hyderabad,14,8,6,0,16,0.245,1,0,0,0,8747666,63483666
90,2017,Sunrisers Hyderabad,14,8,5,1,17,0.599,0,0,0,1,8200500,63014833


In [7]:
# Create a variable for team salary divided by total salaries for that season (relsal).
IPL['relsal']= IPL['salaries']/IPL['allsal']

In [8]:
# Creating a value for win percentage. Defined as wins divided by games with a result (= games played minus games with no result).
IPL['win_pct']=IPL['won'] / (IPL['played']-IPL['noresult'])

In [9]:
# Taking another look at our dataframe with our newly created variables.
IPL

,year,team,played,won,lost,noresult,points,netrunrate,champions,runnersup,third,fourth,salaries,allsal,relsal,win_pct
0,2008,Chennai Super Kings,14,8,6,0,16,-0.192,0,1,0,0,5825000,34105000,0.170796,0.571429
1,2009,Chennai Super Kings,14,8,5,1,17,0.951,0,0,0,0,6765000,33445000,0.202272,0.615385
2,2010,Chennai Super Kings,14,7,7,0,14,0.274,1,0,0,0,4890000,33075000,0.147846,0.500000
3,2011,Chennai Super Kings,14,9,5,0,18,0.443,1,0,0,0,6330000,62210000,0.101752,0.642857
4,2012,Chennai Super Kings,16,8,7,1,17,0.100,0,1,0,0,7900000,59706250,0.132314,0.533333
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
87,2014,Sunrisers Hyderabad,14,6,8,0,12,-0.399,0,0,0,0,9803333,73973332,0.132525,0.428571
88,2015,Sunrisers Hyderabad,14,7,7,0,14,-0.239,0,0,0,0,8104667,65242665,0.124223,0.500000
89,2016,Sunrisers Hyderabad,14,8,6,0,16,0.245,1,0,0,0,8747666,63483666,0.137794,0.571429
90,2017,Sunrisers Hyderabad,14,8,5,1,17,0.599,0,0,0,1,8200500,63014833,0.130136,0.615385


In [10]:
# first we sort the values
IPL.sort_values(by=['team','year'], ascending=True)

,year,team,played,won,lost,noresult,points,netrunrate,champions,runnersup,third,fourth,salaries,allsal,relsal,win_pct
0,2008,Chennai Super Kings,14,8,6,0,16,-0.192,0,1,0,0,5825000,34105000,0.170796,0.571429
1,2009,Chennai Super Kings,14,8,5,1,17,0.951,0,0,0,0,6765000,33445000,0.202272,0.615385
2,2010,Chennai Super Kings,14,7,7,0,14,0.274,1,0,0,0,4890000,33075000,0.147846,0.500000
3,2011,Chennai Super Kings,14,9,5,0,18,0.443,1,0,0,0,6330000,62210000,0.101752,0.642857
4,2012,Chennai Super Kings,16,8,7,1,17,0.100,0,1,0,0,7900000,59706250,0.132314,0.533333
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
87,2014,Sunrisers Hyderabad,14,6,8,0,12,-0.399,0,0,0,0,9803333,73973332,0.132525,0.428571
88,2015,Sunrisers Hyderabad,14,7,7,0,14,-0.239,0,0,0,0,8104667,65242665,0.124223,0.500000
89,2016,Sunrisers Hyderabad,14,8,6,0,16,0.245,1,0,0,0,8747666,63483666,0.137794,0.571429
90,2017,Sunrisers Hyderabad,14,8,5,1,17,0.599,0,0,0,1,8200500,63014833,0.130136,0.615385


In [11]:
# now we create the lagged dependend variable
IPL['wpc_lag'] = IPL.groupby('team')['win_pct'].shift(1)
IPL

,year,team,played,won,lost,noresult,points,netrunrate,champions,runnersup,third,fourth,salaries,allsal,relsal,win_pct,wpc_lag
0,2008,Chennai Super Kings,14,8,6,0,16,-0.192,0,1,0,0,5825000,34105000,0.170796,0.571429,NaN
1,2009,Chennai Super Kings,14,8,5,1,17,0.951,0,0,0,0,6765000,33445000,0.202272,0.615385,0.571429
2,2010,Chennai Super Kings,14,7,7,0,14,0.274,1,0,0,0,4890000,33075000,0.147846,0.500000,0.615385
3,2011,Chennai Super Kings,14,9,5,0,18,0.443,1,0,0,0,6330000,62210000,0.101752,0.642857,0.500000
4,2012,Chennai Super Kings,16,8,7,1,17,0.100,0,1,0,0,7900000,59706250,0.132314,0.533333,0.642857
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
87,2014,Sunrisers Hyderabad,14,6,8,0,12,-0.399,0,0,0,0,9803333,73973332,0.132525,0.428571,0.625000
88,2015,Sunrisers Hyderabad,14,7,7,0,14,-0.239,0,0,0,0,8104667,65242665,0.124223,0.500000,0.428571
89,2016,Sunrisers Hyderabad,14,8,6,0,16,0.245,1,0,0,0,8747666,63483666,0.137794,0.571429,0.500000
90,2017,Sunrisers Hyderabad,14,8,5,1,17,0.599,0,0,0,1,8200500,63014833,0.130136,0.615385,0.571429


In [12]:
# Regression of win_pct on relsal
lm_1 = smf.ols(formula = 'win_pct ~ relsal', data=IPL).fit()
print(lm_1.summary())

                            OLS Regression Results                            
Dep. Variable:                win_pct   R-squared:                       0.002
Model:                            OLS   Adj. R-squared:                 -0.009
Method:                 Least Squares   F-statistic:                    0.1622
Date:                Tue, 04 Aug 2026   Prob (F-statistic):              0.688
Time:                        18:12:09   Log-Likelihood:                 45.259
No. Observations:                  92   AIC:                            -86.52
Df Residuals:                      90   BIC:                            -81.47
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.4752      0.063      7.563      0.0

From the above regression we can see this model is not very good.  We have a R-squared of 0.002 saying that our regression explains 0.2 percent of variation in our data.  We also see that our only dependent variable of relsal has a p-value way higher than 0.05 meaning that it is not significant in our regression.  Next we will try to add the lagged win percentage to this model to see if we can get a stronger outcome.

In [13]:
# Regression of win_pct on relsal and wpc_lag
lm_2 = smf.ols(formula = 'win_pct ~ relsal+wpc_lag', data=IPL).fit()
print(lm_2.summary())

                            OLS Regression Results                            
Dep. Variable:                win_pct   R-squared:                       0.040
Model:                            OLS   Adj. R-squared:                  0.015
Method:                 Least Squares   F-statistic:                     1.591
Date:                Tue, 04 Aug 2026   Prob (F-statistic):              0.210
Time:                        18:17:04   Log-Likelihood:                 43.314
No. Observations:                  79   AIC:                            -80.63
Df Residuals:                      76   BIC:                            -73.52
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.3489      0.087      4.003      0.0

This regression is a little better but still not very good and we fall in to a couple of the same problems.  Firstly, our R-squared is 0.040 meaning our regression exlpains about 4 percent of variation in our data which we would like to be much higher.  Lastly, again, both of our dependent variables have a p-value substantially higher than 0.05 meaning they aare not significant.  Now let's try one final regression adding in the fixed effects for each team.

In [14]:
# Regression of win_pct on relsal, wpc_lag, and team fixed effects
lm_3 = smf.ols(formula = 'win_pct ~ relsal+wpc_lag+C(team)', data=IPL).fit()
print(lm_3.summary())

                            OLS Regression Results                            
Dep. Variable:                win_pct   R-squared:                       0.297
Model:                            OLS   Adj. R-squared:                  0.157
Method:                 Least Squares   F-statistic:                     2.117
Date:                Tue, 04 Aug 2026   Prob (F-statistic):             0.0245
Time:                        18:22:18   Log-Likelihood:                 55.639
No. Observations:                  79   AIC:                            -83.28
Df Residuals:                      65   BIC:                            -50.10
Df Model:                          13                                         
Covariance Type:            nonrobust                                         
                                             coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------------------

This regression shows big improvements over our first two models.  Our R-squared for this regression sits at 0.297 stating that this regression explains 29.7 percent of variation in the data which we would still like to be higher but is a huge increase over our other models.  We can also see that about half of the team fixed effects are significant while the other half along with wpc_lag and relsal again are not significant.

####**Conclusion:**

After doing this lab and running all three regressions we can conclude that cricket, and the IPL in particular, seems to be the only league we studied where relsal and lagged win percentage both seem to have no significant effect on win percentage.  Being that our model is still not as good as we would hope for, since it only explains about 30 percent of variability in our data, a next step could be adding more different dependent variables along with some interactions to see if we can get a better model in terms of R-squared.  Doing this would also allow us to get a clearer picture to if relsal and lagged win percentage will ever be significant values in a regression or if they just don't matter as much as they do in other sports leagues to increase win percentage.